scripts/setup_rdma_env.sh

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+WORKING_DIR="$(readlink -f "$(dirname -- "${BASH_SOURCE[0]}")")/rdma"
+IRDMA_DMID="832291"
+IRDMA_VER="1.15.11"
+
+if [[ -f $WORKING_DIR ]]; then
+    echo "Can't create rdma directory because of the rdma file $WORKING_DIR"
+    exit 1
+fi
+
+mkdir -p "$WORKING_DIR"
+
+install_irdma() {
+    echo "Installing irdma driver..."
+    pushd "$WORKING_DIR"
+    wget "https://downloadmirror.intel.com/${IRDMA_DMID}/irdma-${IRDMA_VER}.tgz" -O ./irdma.tar.gz
+    tar -xf ./irdma.tar.gz
+    pushd "./irdma-${IRDMA_VER}"
+    sudo ./build.sh
+    popd
+    popd
+    modprobe irdma
+}
+
+configure_irdma() {
+    echo "Configuring irdma driver..."
+
+    # enable RoCE
+    roce_ena_val=$(grep "options irdma roce_ena=" /etc/modprobe.d/irdma.conf | cut -d "=" -f 2)
+    if [[ -z "$roce_ena_val" ]]; then
+        echo "options irdma roce_ena=1" | sudo tee -a /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    elif [[ "$roce_ena_val" != "1" ]]; then
+        sudo sed -i '/options irdma roce_ena=/s/roce_ena=[0-9]*/roce_ena=1/' /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    fi
+
+    # increase irdma Queue Pair limit
+    limits_sel_val=$(grep "options irdma limits_sel=" /etc/modprobe.d/irdma.conf | cut -d "=" -f 2)
+    if [[ -z "$limits_sel_val" ]]; then
+        echo "options irdma limits_sel=5" | sudo tee -a /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    elif [[ "$limits_sel_val" != "5" ]]; then
+        sudo sed -i '/options irdma limits_sel=/s/limits_sel=[0-9]*/limits_sel=5/' /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    fi
+}
+
+install_perftest() {
+    echo "Installing perftest dependencies..."
+    sudo apt update
+    sudo apt install -y libibverbs-dev librdmacm-dev libibumad-dev libpci-dev
+    echo "Installing perftest..."
+    pushd "$WORKING_DIR"
+    wget https://github.com/linux-rdma/perftest/releases/download/24.07.0-0.44/perftest-24.07.0-0.44.g57725f2.tar.gz -O perftest.tar.gz
+    tar -xf perftest.tar.gz
+    pushd ./perftest-24.07.0
+    ./autogen.sh
+    ./configure
+    make -j$(nproc)
+    sudo make install
+    popd
+    popd
+}
+
+install_libfabric() {
+    echo "Installing libfabric from the source..."
+    pushd "$WORKING_DIR"
+    git clone --depth 1 --branch v1.22.0 https://github.com/ofiwg/libfabric libfabric
```

Comment:
This is done wrongly - this will download the libfabric to your working directory (place from where you will call this script). It is also better to use the `wget` or `curl` and download it as `tar.gz` or just have it as a repository module.

`git clone --depth 1 --branch v1.22.0 https://github.com/ofiwg/libfabric ${WORKING_DIR}/libfabric`

or
```bash
. ${REPOSITORY_DIR}/common.sh

# GITHUB_CREDENTIALS="username:password"
# URL construction: https://${GITHUB_CREDENTIALS}@github.com/${name}/archive/${version}.tar.gz
# $1 - name
# $2 - version
# $3 - dest_dir
function git_download_strip_unpack()
{
    # Version can be commit sha or tag, examples:
    # version=d2515b90cc0ef651f6d0a6661d5a644490bfc3f3
    # version=refs/tags/v${JPEG_XS_VER}
    name="${1}"
    version="${2}"
    dest_dir="${3}"
    filename="$(get_filename ${version})"
    [ -n "${GITHUB_CREDENTIALS}" ] && creds="${GITHUB_CREDENTIALS}@" || creds=""

    mkdir -p "${dest_dir}"
    curl -Lf https://${creds}github.com/${name}/archive/${version}.tar.gz -o "${dest_dir}/${filename}.tar.gz"
    tar -zx --strip-components=1 -C "${dest_dir}" -f "${dest_dir}/${filename}.tar.gz"
    rm -f "${dest_dir}/${filename}.tar.gz"
}

LIBFABRIC_VER="v1.22.0"
git_download_strip_unpack "ofiwg/libfabric" "${LIBFABRIC_VER}" "${LIBFABRIC_DIR}"
```